src/applications/disability-benefits/all-claims/tests/utils/clean-toxic-exposure-data.unit.spec.jsx

```diff
@@ -0,0 +1,1466 @@
+        'chemical',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.have.property(
+        'water',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.have.property(
+        'mustardgas',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.have.property(
+        'radiation',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.have.property(
+        'other',
+      );
+
+      // Should not have details for false selections
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'chromium',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'depleted',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'mos',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'shad',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'shipyard',
+      );
+    });
+  });
+
+  describe('comprehensive edge cases for all toxic exposure keys', () => {
+    it('should remove all details when main selection is missing', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          // Missing gulfWar1990 but has details
+          gulfWar1990Details: {
+            bahrain: { startDate: '1990-01-01', endDate: '1990-12-31' },
+          },
+          // Missing herbicide but has details
+          herbicideDetails: {
+            vietnam: { startDate: '1968-01-01', endDate: '1970-01-01' },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure).to.not.have.property('gulfWar1990Details');
+      expect(result.toxicExposure).to.not.have.property('herbicideDetails');
+    });
+
+    it('should handle all toxic exposure keys from getAllToxicExposureKeys', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { none: true },
+          gulfWar1990: { bahrain: true },
+          gulfWar1990Details: { bahrain: { startDate: '1990-01-01' } },
+          gulfWar2001: { afghanistan: true },
+          gulfWar2001Details: { afghanistan: { startDate: '2001-01-01' } },
+          herbicide: { vietnam: true },
+          herbicideDetails: { vietnam: { startDate: '1968-01-01' } },
+          otherHerbicideLocations: { description: 'Thailand' },
+          otherExposures: { burnpits: true, asbestos: true },
+          otherExposuresDetails: {
+            burnpits: { startDate: '2003-01-01' },
+            asbestos: { startDate: '1980-01-01' },
+          },
+          specifyOtherExposures: 'Lead paint',
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      // When 'none' is selected, only conditions should remain
+      expect(result.toxicExposure).to.deep.equal({
```

Comment:
What is the result when other conditions with **false** or **true** values are present, in addition to `none: true`?